pyat/at/collective/wake_object.py

```diff
@@ -165,3 +165,49 @@ def reswall(self, wcomp, length, rvac, conduct, beta, yokoya_factor=1):
                                       length, rvac, conduct, beta)
         else:
             raise AtError('Invalid WakeComponent: {}'.format(wcomp))
+
+
+def res_object(srange, wakecomp, frequency, qfactor, rshunt, beta,
```

Comment:
I like the fact that it allows for multiple resonators. When the WakeComponent is the same for all resonators (for example I make a WakeZ which sums many resonators of different Q, Rs, fr) the interface requires [WakeComponent.Z]*10 which is not so clean. Maybe it can be added that if WakeComponent is not an array, then it is assumed for all resonators. 

```
numRes = 10
p = res_object(srange, [WakeComponent.Z]*numRes, np.arange(1,numRes+1)*1e9, np.ones(numRes)*5000, np.arange(numRes)*1e4, 1, yokoya_factor=1)

```

Reply:
@lcarver I understand your point but I would have to think about this one. I am not quite sure this can be easily and cleanly handled

Reply:
@lcarver, so I have thought about it and what I can provide is an optional `nelems=1` argument. All input arguments would therefore have to be broadcastable to `(nelem,)`
This also presents any conflicts with `wake_elements.py` where the default of nelems=1 would be used